llvm/lib/Target/EraVM/EraVMInstrInfo.td

```diff
@@ -1248,3 +1259,95 @@ def : Pat<(EraVMTrap), (PANICr R0)>;
 
 def : Pat<(EraVMselectcc GRPTR:$rs0, GRPTR:$rs1, imm:$cc),
           (FATPTR_SELrrr GRPTR:$rs0, GRPTR:$rs1, imm:$cc)>;
+
+// ============================ Overflow ADDs =====================================
```

Comment:
Patterns for `large_imm` are missing.

Reply:
Hi @vladimirradosavljevic , thanks for your comments. I checked the `defm ADD  : ArithICommutable<OpAdd, add>;` and found there is no case using `large_imm`. So it looks to me I can't expand ADD_V with `large_imm` with ADD_xxx with `large_imm ` directly unless I add `large_imm` case into `ArithICommutable`. Is this understanding correct?

Reply:
Hi @vladimirradosavljevic , I create a separate [PR](https://github.com/matter-labs/era-compiler-llvm/pull/550) for this item

> Remove all tie constraints from select instructions from EraVMInstrInfo.td and move tie handling to EraVMTieSelectOperands pass. Create a separate PR for this to check the numbers, since I expect we will get some differences, as now we would only tie killed input registers with output.

Reply:
`large_imm` is placed into constant pool, so you will have to do something like this:
```
def : Pat<(EraVMAdd_v GR256:$rs0, large_imm:$rs1),
          (ADDcrr_v (constant_pool imm:$rs1), 0, GR256:$rs0, 0)>;
```

Reply:
`large_imm` is added now just like `imm16` .